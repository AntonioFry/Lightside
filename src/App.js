import React, { Component } from 'react';
import Header from './Header/Header';
import Aside from './Aside/Aside';
import NavBar from './NavBar/NavBar';
import DataContainer from './DataContainer/DataContainer';
import peopleData from './mockData/peopleData';
import filmData from './mockData/filmData';

class App extends Component {
  constructor() {
    super()
    this.state = {
      openingCrawl: undefined
    }
  }

componentDidMount() {
  const randomNumber = Math.floor(Math.random() * 7)
  const randomOpener = filmData.results.splice(randomNumber, 1)[0].opening_crawl
  this.setState({ openingCrawl: randomOpener })
}



  // componentDidMount() {
  //   this.fetchApi('https://swapi.co/api/people/');
  // }

  // fetchApi = (apiToFetch) => {
  //   fetch(apiToFetch)
  //     .then(result => result.json())
  //     .then((entries) => {
  //         console.log(entries.results)
  //         })
  //     .catch((error) => console.log(error));
  // }

  render() {
    return (
      <main className="App">
        <Aside openingCrawl={this.state.openingCrawl}/>
        <section className='app-right-side'>
          <Header />
          <NavBar />
          <DataContainer />
        </section>
      </main>
    );
  }
}

export default App;
