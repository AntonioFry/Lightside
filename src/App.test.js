import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from 'enzyme';

describe('App', () => {

  var wrapper;
  beforeEach(function() {
    wrapper = shallow(<App />);
  });

  it('should match the snapshot with the data being passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // it('should update state when fetchData is invoked', () => {

  //   wrapper.instance().fetchData('people');

  //   expect(wrapper.state('people').length).toEqual(10);
  // });

})
