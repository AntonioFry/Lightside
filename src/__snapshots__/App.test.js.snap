// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with the data being passed in correctly 1`] = `ShallowWrapper {}`;
